src/settings/panel.tsx

```diff
@@ -0,0 +1,245 @@
+import type { FieldProps } from '@rjsf/utils';
+import validator from '@rjsf/validator-ajv8';
+import { JSONSchema7 } from 'json-schema';
+import React from 'react';
+
+import { IDict, instructions } from './instructions';
+import baseSettings from './schemas/base.json';
+import ProviderSettings from './schemas';
+
+const MD_MIME_TYPE = 'text/markdown';
+const STORAGE_NAME = '@jupyterlite/ai:settings';
+const INSTRUCTION_CLASS = 'jp-lite-ai-settings-instruction';
+
+export const aiSettingsRenderer = (options: {
+  rmRegistry?: IRenderMimeRegistry;
+}): IFormRenderer => {
+  return {
+    fieldRenderer: (props: FieldProps) => {
+      props.formContext = { ...props.formContext, ...options };
+      return <AiSettings {...props} />;
+    }
+  };
+};
+
+export interface ISettingsFormStates {
+  schema: JSONSchema7;
+  instruction: HTMLElement | null;
+}
+
+const WrappedFormComponent = (props: any): JSX.Element => {
+  return <FormComponent {...props} validator={validator} />;
+};
+
+export class AiSettings extends React.Component<
+  FieldProps,
+  ISettingsFormStates
+> {
+  constructor(props: FieldProps) {
+    super(props);
+    this._rmRegistry = props.formContext.rmRegistry ?? null;
+    this._settings = props.formContext.settings;
+
+    // Initialize the providers schema.
+    const providerSchema = JSONExt.deepCopy(baseSettings) as any;
+    providerSchema.properties.provider = {
+      type: 'string',
+      title: 'Provider',
+      description: 'The AI provider to use for chat and completion',
+      default: 'None',
+      enum: ['None'].concat(Object.keys(ProviderSettings))
+    };
+    this._providerSchema = providerSchema as JSONSchema7;
+
+    // Check if there is saved values in local storage, otherwise use the settings from
+    // the setting registry (led to default if there are no user settings).
+    const storageSettings = localStorage.getItem(STORAGE_NAME);
+    if (storageSettings === null) {
+      const labSettings = this._settings.get('AIprovider').composite;
+      if (labSettings && Object.keys(labSettings).includes('provider')) {
+        // Get the provider name.
+        const provider = Object.entries(labSettings).find(
+          v => v[0] === 'provider'
+        )?.[1] as string;
+        // Save the settings.
+        const settings: any = {
+          _current: provider
+        };
+        settings[provider] = labSettings;
+        localStorage.setItem(STORAGE_NAME, JSON.stringify(settings));
+      }
+    }
+
+    // Initialize the settings from the saved ones.
+    this._provider = this.getCurrentProvider();
+    this._currentSettings = this.getSettings();
+
+    // Initialize the schema.
+    const schema = this._buildSchema();
+    this.state = { schema, instruction: null };
+
+    this._renderInstruction();
+
+    // Update the setting registry.
+    this._settings
+      .set('AIprovider', this._currentSettings)
+      .catch(console.error);
+  }
+
+  async _renderInstruction(): Promise<void> {
```

Comment:
Maybe this method can be kept private for now?

```suggestion
  private async  _renderInstruction(): Promise<void> {
```